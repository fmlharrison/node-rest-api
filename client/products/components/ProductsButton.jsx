import React, { Component } from "react";
import PropTypes from "prop-types";

class ProductsButton extends Component {
  handleClick = event => {
    event.preventDefault();
    return this.props.showProducts();
  };

  render() {
    return <button onClick={this.handleClick}>Show Products</button>;
  }
}

ProductsButton.propTypes = {
  showProducts: PropTypes.func.isRequired
};

export default ProductsButton;
