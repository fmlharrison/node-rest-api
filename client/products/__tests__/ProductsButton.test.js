import React from "react";
import ProductsButton from "../components/ProductsButton.jsx";

test("Renders button component", () => {
  const component = shallow(<ProductsButton showProducts={jest.fn()} />);
  expect(component).toMatchSnapshot();
});

test("Clicking button fires props callback", () => {
  const mockPropFunc = jest.fn();
  const component = shallow(<ProductsButton showProducts={mockPropFunc} />);
  component.find("#show-products").simulate("click", { preventDefault() {} });
  expect(mockPropFunc.mock.calls.length).toBe(1)
});
