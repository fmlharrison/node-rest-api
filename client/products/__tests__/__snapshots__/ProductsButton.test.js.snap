// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders button component 1`] = `
<button
  id="show-products"
  onClick={[Function]}
>
  Show Products
</button>
`;
